django/db/models/query.py

```diff
@@ -291,10 +291,11 @@ def __getitem__(self, k):
                 'QuerySet indices must be integers or slices, not %s.'
                 % type(k).__name__
             )
-        assert ((not isinstance(k, slice) and (k >= 0)) or
-                (isinstance(k, slice) and (k.start is None or k.start >= 0) and
-                 (k.stop is None or k.stop >= 0))), \
-            "Negative indexing is not supported."
+
+        if ((not isinstance(k, slice) and (k < 0)) or
```

Comment:
Above it is just ensured that `k` is an int or slice. What do you think--every little bit helps with this dense condition?

```suggestion
        if ((isinstance(k, int) and k < 0) or
```